server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/InternalGeoHashGrid.java

```diff
@@ -63,26 +65,29 @@
          * Read from a stream.
          */
         private Bucket(StreamInput in) throws IOException {
+            type = GeoHashType.readFromStream(in);
```

Comment:
We need to check the version here

Reply:
The version is checked inside the type itself -- https://github.com/elastic/elasticsearch/pull/30320/files#diff-75ad5ae5ac78babf26799166a287cfe1R51

Reply:
We should do the version check here IMO. The way to serialise the type is not version dependant (since the type is new), but the fact that the type is serialised is version dependant since previous versions do not know about the concept of a type here.

Reply:
Again I think this is a readability thing for me. If I come across the code in GeohashGridAggregationBuilder I think that we have a bug because the type is always serialised. I then have to dig into this class to find out that sometimes, depending on the version it doesn't actually serialise itself at all. To me this seem weird and for the sake of avoiding having 4 checks in favour of 2 I don't think its worth the confusion potential.